### PR TITLE
added yahoo xss-filters

### PR DESCRIPTION
**Pull request by davetayls**
this is from https://github.com/yahoo/xss-filters/wiki

### CODE REVIEW

**Comment by vvakame** [comment]
*xss-filters/xss-filters.d.ts*

check list

* [ ] is collect [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)?
  * https://www.npmjs.com/package/xss-filters
  * http://bower.io/search/?q=xss-filters
  * others?
* [X] has a [test file](http://definitelytyped.org/guides/contributing.html#tests)? (xss-filters/xss-filters-tests.ts or others)
* [ ] pass the Travic-CI test?

File: xss-filters/xss-filters.d.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  uriComponentInSingleQuotedAttr(s:string):string;
+  uriComponentInUnQuotedAttr(s:string):string;
+  uriFragmentInHTMLComment(s:string):string;
+  uriFragmentInHTMLData(s:string):string;
+  uriFragmentInDoubleQuotedAttr(s:string):string;
+  uriFragmentInSingleQuotedAttr(s:string):string;
+  uriFragmentInUnQuotedAttr(s:string):string;
+}
+
+declare var xssFilters:XSSFilters;
```
Comment (vvakame) [inline xss-filters/xss-filters.d.ts]: please move this line into `declare module`. see https://github.com/borisyankov/DefinitelyTyped/blob/master/gulp/gulp.d.ts#L281-L284

Comment (davetayls) [inline xss-filters/xss-filters.d.ts]: only issue with this is that when you include this in the browser with a script tag it creates a global `xssFilters` namespace on the window object which is why I included this statement

Comment (vvakame) [inline xss-filters/xss-filters.d.ts]: oh, sorry. It is my mistake. :wink:

**Comment by vvakame** [comment]
@davetayls thanks mate!

**Pull request merged**